lib/pbench/server/database/models/users.py

```diff
@@ -51,14 +63,16 @@ def get_protected():
         return ["registered_on", "id"]
 
     @staticmethod
-    def query(id=None, username=None, email=None):
+    def query(id=None, username=None, email=None, all=False):
         # Currently we would only query with single argument. Argument need to be either username/id/email
         if username:
             user = Database.db_session.query(User).filter_by(username=username).first()
         elif id:
             user = Database.db_session.query(User).filter_by(id=id).first()
         elif email:
             user = Database.db_session.query(User).filter_by(email=email).first()
+        elif all:
+            return Database.db_session.query(User).all()
```

Comment:
Any particular reason not to assign the value to be returned to `user`, like all the other cases?  (Alternately, if we like abrupt returns, any reason not to use them in the other cases?)

Reply:
Other assigns are singular meaning we would return 1 user, but with `all` we return all the users

Reply:
While Python allows this, it's difficult to describe and semantically muddy. Think of the return type hint. You can't use `-> User`, for example, because it's not. The best you could do is `-> Any`, which isn't very useful. I don't like overloading the return to "single entity or collection of entities" and I think we should have a `query-one` and `query-list` distinction with several methods. In fact, the `query-one` variant could be a simple wrapper around the `query-list` that returns `.first()` ... and maybe should raise an exception if there's more than one, since the caller explicitly specified the expectation that there'd be only one. (If it really wants just the first and is willing to silently ignore others, it can just call `query-list` directly and do the `.first()` itself.)

Reply:
Fixed

Reply:
I agree 100%, and this is what I was alluding to when when I asked for the return type hint.  :-D

> The best you could do is `-> Any`

I will quibble with this statement, though (so, I guess it's not "100%"...).  You _could_ have done `-> Union[User, list[User]]` (or similar), but, as Dave indicated, just because Python will let you do that doesn't mean you _should_.